### ISSUE TITLE
BUG: Use of constant in `tests/slate/test_hdg_poisson.py::test_hdg_convergence` causes errors

### ISSUE THREAD

**Issue by JDBetteridge** [firedrakeproject/firedrake#3802]
**Describe the bug**
There is something strange happening in the `tests/slate/test_hdg_poisson.py::test_hdg_convergence` test if run in parallel, but not in the "default pytest order". This is seen in the test suite cleanup PR https://github.com/firedrakeproject/firedrake/pull/3385 when using `pytest-split`:

https://github.com/firedrakeproject/firedrake/actions/runs/11355091421/job/31583739399?pr=3385

This error goes away if a function in the real space is used

**Steps to Reproduce**
This is quite difficult you have to run
```
pytest \
    --splitting-algorithm least_duration \
    --splits 4 \
    --group ??? \
    --timeout=1800 \
    --timeout-method=thread \
    -o faulthandler_timeout=1860 \
    -m "parallel[3] and not broken" \
    -v tests
```
With `???` replaced with 1 to 4 and one of the runs should fail. Reproducing the error locally has proven tricky though.

**Expected behavior**
This test should pass with a `Constant` it seems something in some part of the test suite is throwing off the numbering :confused: 

**Error message**
https://github.com/firedrakeproject/firedrake/actions/runs/11355091421/job/31583739399?pr=3385

**Environment:**
 - CI

**Additional Info**

**Comment by connorjward** [comment]
Can you share the current "fix" for this so it's clear where I need to look?

**Comment by wence-** [comment]
This line https://github.com/firedrakeproject/firedrake/blob/6cc4ff80915fee17522b592d7b16e505714432ad/tests/slate/test_hdg_poisson.py#L53

wants to be a function in Real, I suspect.

My guess would be `Constant`s are not renumbered correctly by UFL and one process gets out of sync?

**Comment by JDBetteridge** [comment]
The fix was in this commit: https://github.com/firedrakeproject/firedrake/pull/3385/commits/911b0ade6afd859511e4d50084546fb1e2e85412

**Comment by connorjward** [comment]
Fixed in #3808. We were hardcoding constant names in SLATE (which I wrote some time ago).
